src/Tools/dotnet-monitor/CapabilitiesPostConfigureOptions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    public abstract class CapabilityPostConfigureOptions<T> : IPostConfigureOptions<T> where T : class
+    {
+        private MonitorCapability? _capability;
+
+        public CapabilityPostConfigureOptions(MonitorCapability? capability)
+        {
+            _capability = capability;
+        }
+
+        public abstract void PostConfigure(string? name, T options);
```

Comment:
I think this is ok, but just to offer some additional options for consideration:
- Override seal PostConfigure and have a abstract method similiar to `abstract bool GetIsEnabled(T options). The base class PostConfigure implementation can call GetIsEnabled and PostConfigure(bool) in the implementation.
- Alternatively, Make CapabilityPostConfigureOptions<T> non-abstract and have it take an additonal parameter of Func<T, bool>. During registration this might look something like `RegisterCapability<CallStackOptions>( (CallStackOptions options) => options.Enabled`